load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,124 +15,222 @@
-        builder =
-            builder
-                .cpsProject(cpsArgs.getProject())
-                .numResponseThreads(cpsArgs.getNumResponseThreads())
-                .rateLimit(cpsArgs.getRateLimit());
-        if (baseArgs.isPublish()) {
-          // Create a task which publishes to CPS.
-          taskArgs =
-              builder
-                  .messageSize(baseArgs.getMessageSize())
-                  .batchSize(cpsArgs.getBatchSize())
-                  .build();
-          cpsHandler.setLatencyType(MessageProcessingHandler.LatencyType.PUB_TO_ACK);
-          CPSRoundRobinPublisher publisher = new CPSRoundRobinPublisher(cpsArgs.getNumClients());
-          log.info("Creating a task which publishes to CPS.");
-          new CPSPublishingTask(taskArgs, publisher, cpsHandler).execute();
-        } else {
-          // Create a task which consumes from CPS.
-          if (baseArgs.isDumpData()) {
-            cpsHandler.setFiledump(new File(Utils.CPS_FILEDUMP_PATH));
-          }
-          taskArgs = builder.build();
-          cpsHandler.setLatencyType(MessageProcessingHandler.LatencyType.END_TO_END);
-          CPSRoundRobinSubscriber subscriber = new CPSRoundRobinSubscriber(cpsArgs.getNumClients());
-          log.info("Creating a task which consumes from CPS.");
-          new CPSSubscribingTask(taskArgs, subscriber, cpsHandler).execute();
-        }
-      } else {
-        // The "kafka" command was invoked.
-        MessageProcessingHandler kafkaHandler = new MessageProcessingHandler(baseArgs.getNumMessages());
-        kafkaHandler.setLatencyType(MessageProcessingHandler.LatencyType.PUB_TO_ACK);
-        builder = builder.broker(kafkaArgs.getBroker());
-        if (baseArgs.isPublish()) {
-          // Create a task that publishes to Kafka.
-          taskArgs = builder.messageSize(baseArgs.getMessageSize()).build();
-          KafkaProducer<String, String> publisher =
-              KafkaPublishingTask.getInitializedProducer(taskArgs);
-          log.info("Creating a task which publishes to Kafka");
-          new KafkaPublishingTask(taskArgs, publisher, kafkaHandler).execute();
-        } else {
-          // Create a task that consumes from Kafka.
-          if (baseArgs.isDumpData()) {
-            kafkaHandler.setFiledump(new File(Utils.KAFKA_FILEDUMP_PATH));
-          }
-          taskArgs = builder.broker(kafkaArgs.getBroker()).build();
-          kafkaHandler.setLatencyType(MessageProcessingHandler.LatencyType.END_TO_END);
-          KafkaConsumer<String, String> consumer =
-              KafkaConsumerTask.getInitializedConsumer(taskArgs);
-          log.info("Creating a task which consumes from Kafka.");
-          new KafkaConsumerTask(taskArgs, consumer, kafkaHandler).execute();
+      Client.messageSize = messageSize;
+      Client.requestRate = 1;
+      Client.startTime = Timestamp.newBuilder().setSeconds(System.currentTimeMillis() / 1000 + 90).build();
```

Comment:
Currently it is just a safe default, since 90 seconds is enough time to guarantee everything has started with a 30 second or so buffer.